plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightClientManager.java

```diff
@@ -128,40 +145,39 @@ private void buildClientAndAddToPool(Location location, DiscoveryNode node) {
             );
             return;
         }
-        flightClients.computeIfAbsent(node.getId(), key -> buildClient(location));
+        if (closed) {
+            return;
+        }
+        FlightClient flightClient = buildClient(location);
+        flightClients.put(node.getId(), new ClientHolder(location, flightClient));
     }
 
     private void requestNodeLocation(String nodeId, CompletableFuture<Location> future) {
         NodesFlightInfoRequest request = new NodesFlightInfoRequest(nodeId);
-        try {
-
-            client.execute(NodesFlightInfoAction.INSTANCE, request, new ActionListener<>() {
-                @Override
-                public void onResponse(NodesFlightInfoResponse response) {
-                    NodeFlightInfo nodeInfo = response.getNodesMap().get(nodeId);
-                    if (nodeInfo != null) {
-                        TransportAddress publishAddress = nodeInfo.getBoundAddress().publishAddress();
-                        String address = publishAddress.getAddress();
-                        int flightPort = publishAddress.address().getPort();
-                        Location location = clientConfig.sslContextProvider != null
-                            ? Location.forGrpcTls(address, flightPort)
-                            : Location.forGrpcInsecure(address, flightPort);
-
-                        future.complete(location);
-                    } else {
-                        future.completeExceptionally(new IllegalStateException("No Flight info received for node: [" + nodeId + "]"));
-                    }
-                }
+        client.execute(NodesFlightInfoAction.INSTANCE, request, new ActionListener<>() {
```

Comment:
Why we removed `try` from here?

Reply:
this is not intentional and I plan to revert it but missed, thanks for catching it.